### PR TITLE
perf: eliminate blocking preference access

### PR DESCRIPTION
**Pull request by LUC4N3X**
## Summary

`LevyraPreferences` wrapped every getter and setter in `runBlocking(Dispatchers.IO)`. Switching the dispatcher does not help: the caller, often the main thread during ViewModel, service or UI work, still parks until DataStore finishes. This PR serves preference reads from a single in-memory snapshot and writes to DataStore through one ordered background writer.

The same PR also fixes Home first-launch issues found during device testing: Quick Picks was empty on a fresh install, artists took a long time to appear, the player opened in artwork mode instead of Canvas, and Levyra Collections cards shared similar colours.

## What changed

- New process-wide `LevyraPreferencesStore`. It loads DataStore once, keeps the latest `Preferences` in memory, applies edits to memory synchronously and persists them in FIFO order through a single writer coroutine in an owned `SupervisorJob + Dispatchers.IO` scope.
- `LevyraPreferences` getters read the snapshot. Structured values (full snapshot, audio settings with custom presets, home sections, albums, chart and orbit caches) are parsed once per snapshot. `automationSettingsFlow` now comes from the same snapshot instead of a second DataStore collector.
- Suspend writers (`restoreSnapshot`, `setMotionArtworkEnabled`, `setAutomationSettings`, `persistBackupTreeUri`) still wait for durable persistence. Writes issued before the first load are applied on top of the persisted state, so defaults never overwrite stored values.
- `LevyraApplication` starts the initial load in `onCreate`.
- Player visual mode now defaults to immersive Canvas on fresh installs. Stored choices and the legacy motion-artwork mapping are unchanged.
- Quick Picks: startup seeds (`chart-seed-*` ids) are now accepted. The seed pool covers the language catalog plus global hits, orbit tracks stay excluded, and the shelf targets 20 tracks with no duplicates. Seeds live in UI state, so the existing official-artwork queue replaces YouTube video frames with album covers.
- Quick Picks layout: a horizontal pager of 2×4 compact cards, with actions on long-press.
- Home artists: with nothing cached, loading starts after the short grace delay instead of the full startup delay, and each resolved batch is published.
- Levyra Collections: each card gets a distinct colour pair, and the it/en collection titles are clearer.

## Type of change

- [x] Bug fix
- [x] New feature
- [x] Performance improvement
- [ ] Refactor or maintenance
- [x] UI or UX change
- [ ] Localization or accessibility
- [ ] Build, CI, packaging, or release change
- [ ] Documentation
- [ ] Breaking change

## Scope

- **Platform:** Android
- **Area:** Player / UI / System integration

### Automated checks

- [ ] Android: `./gradlew --no-daemon :app:lintRelease :app:testReleaseUnitTest :app:assembleRelease`
- [ ] Desktop: `cd desktop && ./gradlew check assemble`
- [x] Targeted tests were added or updated
- [ ] Not applicable, with the reason explained below

Checks that were actually run: `./gradlew :app:testDebugUnitTest` (2014 tests, 0 failures), `./gradlew :app:assembleDebug`, and `python scripts/ai_quality_gate.py --profile fast` (passed). The release lint/test/assemble tasks were not run locally; CI covers them.

New `LevyraPreferencesStoreTest` covers: default snapshot, persisted values and legacy mappings loading into memory, immediate visibility of writes, survival across recreation, 200 rapid writes keeping the last intent, concurrent writes to different keys, writes issued before the initial load, backup restore visibility and durability (custom EQ preset, automation, interface, Jam name), parse-once behaviour of structured values, and the automation flow following in-memory writes.

### Manual testing

All device testing ran on a physical Samsung SM-S936B over ADB Wireless Debugging, not an emulator and not USB.

| Environment | Scenario | Result |
|---|---|---|
| SM-S936B, debug, Wi-Fi ADB | Cold launch, warm launch, onboarding | Pass |
| SM-S936B | Search, playback start, pause/resume, next, previous, Now Playing | Pass |
| SM-S936B | Settings changed (theme Neon Cyan, dynamic colour off, Canvas quality High, visual mode, background Blur, haptics off), force-stop and relaunch | Settings persisted |
| SM-S936B | Background and foreground, last track restored after relaunch | Pass |
| SM-S936B | Fresh install: player visual mode defaults to immersive Canvas | Pass |
| SM-S936B | Fresh install: Quick Picks shows 20 non-orbit tracks in 2×4 horizontal pages | Pass |
| SM-S936B | Logcat during the flows | No crash, ANR or DataStore error from the app |
| SM-S936B | Memory, 3–4 cold starts each, `main` vs this branch | Java heap 40–70 MB vs 38–72 MB. PSS differences come from the Graphics category, which varies similarly on both builds |

Language switching was not exercised on the device.

## Risk and compatibility

- **Risk level:** Medium
- **Compatibility or migration notes:** Preference keys, defaults (except the player visual mode for users with no stored choice), migrations and the backup format are unchanged.
- **Known limitations:** A synchronous read made before the initial DataStore load finishes waits once for that single shared load; `LevyraApplication` starts the load early so the wait is normally negligible. Fire-and-forget setters persist asynchronously, so a write made milliseconds before the process is killed can be lost. If DataStore fails during a backup restore, the restore reports the error, but the in-memory values stay restored until the next process start. Quick Picks seed covers are resolved over the network, so a video frame can show briefly on first launch.
- **Rollback plan:** Revert this PR

## Reviewer notes

- `LevyraPreferencesStore`: memory updates and write enqueueing happen under one lock so memory and disk ordering match. No disk I/O runs under the lock. No DataStore collector runs after the initial load: this process is the only writer, so memory is authoritative.
- `buildQuickPicks` now accepts `chart-seed-*` ids. Orbit exclusion and per-artist limits are unchanged.
- Performance: the PR removes the identified synchronous preference blocking paths. `am start -W` cold `TotalTime` on the debug build was about 1.6 s on this branch versus 2.7–2.9 s on `main` (small sample, debug build). Treat it as indicative, not a benchmark.

## Release note

Faster, non-blocking settings; Quick Picks ready from the first launch; the player opens in Canvas by default.

## Related issues

- Closes #
- Related to #

## Checklist

- [ ] The PR is focused and contains no unrelated changes
- [x] The implementation follows the existing architecture and naming conventions
- [x] Threading, lifecycle, cancellation, and resource cleanup were reviewed where relevant
- [x] Tests, documentation, localization, and screenshots were updated where required
- [x] No secrets, keystores, generated packages, archives, or local-only files were committed
- [x] Android and Desktop versioning and release channels remain independent
- [ ] CI is green, or every remaining failure is explained in this PR
- [ ] The complete Levyra PR template is preserved and the final description passed through `levyra-humanizer` without changing its claims

## Summary by CodeRabbit

* **New Features**
  * Quick access now offers up to 20 unique tracks in paged, two-column layouts.
  * Track cards include artwork, visual backgrounds, current-track indicators, and long-press actions.
  * Quick access content refreshes with the selected app language and enriched track artwork.
  * The default player visual mode is now Canvas Immersive.

* **Improvements**
  * Home artist shelves update more consistently as results become available.
  * Updated English and Italian labels improve clarity across editorial collections.

### CODE REVIEW

File: app/src/main/java/com/luc4n3x/levyra/data/LevyraPreferencesStore.kt [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+    init {
+        scope.launch {
+            publishInitial(readPersisted())
+            for (write in pendingWrites) persist(write)
+        }
+    }
+
+    fun current(): Preferences {
+        state.value?.let { return it }
+        initialLoad.await()
```
[the remark this answers is not in the document]
Comment (LUC4N3X) [reply app/src/main/java/com/luc4n3x/levyra/data/LevyraPreferencesStore.kt:58]: Not changing this path: the cited `backupSettings()` access is already inside `startupScope.launch` (`SupervisorJob() + Dispatchers.Default`) after the 1.8 s delay, so it does not block `Application.onCreate()` on the main thread. Returning defaults before the initial DataStore load would introduce a real startup regression by exposing transiently incorrect persisted settings.

File: app/src/main/java/com/luc4n3x/levyra/viewmodel/LevyraViewModel.kt [hunk 2]
```diff
@@ -1885,6 +1892,12 @@ class LevyraViewModel(application: Application) : AndroidViewModel(application)
                             resolved.putIfAbsent(hit.browseId.lowercase(), hit)
                         }
                     }
+                    if (!freezeVisibleShelf && resolved.isNotEmpty() && homeArtistsFingerprint == fingerprint) {
+                        val partialArtists = resolved.values.take(HOME_ARTIST_SHELF_SIZE)
+                        _state.update { current ->
+                            if (current.languageCode == languageCode) current.copy(homeArtists = partialArtists) else current
+                        }
```
[the remark this answers is not in the document]
Comment (LUC4N3X) [reply app/src/main/java/com/luc4n3x/levyra/viewmodel/LevyraViewModel.kt]: Fixed. Partial artist updates now merge newly resolved artists with the validated cached shelf, deduplicate by browse ID, and only then apply the shelf limit.
